<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="keywords" content="UCSF, University of California San Francisco, CBC, Computational Biology Core, HDFCCC, Helen Diller Family Comprehensive Cancer Center">
  <meta name="author" content="Henrik Bengtsson">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <link rel="shortcut icon" href="../assets/ico/favicon.ico">
  <title title="UCSF Computational Biology Core (CBC): Software and Computing Resources">CBC: Software and Computing Resources</title>
  <link href="../assets/css/ucsf-banners.css" rel="stylesheet">
  <link href="../assets/css/bootstrap.min.css" rel="stylesheet">
  <link href="../assets/css/cbc-tweaks.css" rel="stylesheet">
</head>
<body>
<nav class="navbar navbar-default navbar-fixed-top" role="navigation" id="topnavbar">
  <div class="top-banner logo">
    <div class="inner">
      <div>
        <div class="logo"></div>
        <a class="logotype" href="http://www.ucsf.edu/">
          <span>University of California San Francisco</span>
        </a>
      </div>
      <div class="nav">
        <ul class="top-banner-menu">
          <li><a href="http://www.ucsf.edu/about">About UCSF</a></li>
          <li><a href="http://www.ucsf.edu/search">Search UCSF</a></li>
          <li><a href="http://www.ucsfhealth.org/">UCSF Medical Center</a></li>
        </ul>
      </div>
    </div>
  </div> 
  <div class="container-fluid">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html" title="Computational Biology Core">CBC</a>
    </div>

    <div class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="../people/index.html">People</a></li>
        <li><a href="../consultation/index.html">Consultation</a></li>
        <li><a href="../sequencing/index.html">Sequencing</a></li>
        <li><a href="../software/index.html">Software</a></li>
        <li><a href="../stories/index.html">Stories</a></li>
        <li><a href="../education/index.html">Education</a></li>
        <li><a href="../publications/index.html">Publications</a></li>
        <li><a href="../contact/index.html">Contact</a></li>
      </ul>
    </div>
  </div>
</nav>

<div id="wrap">

<div id="sublogocontainer" style="visibility: hidden;">
 <a href="http://cancer.ucsf.edu/"
    title="UCSF Helen Diller Family Comprehensive Cancer Center">
   <img id="sublogo"
        src="../assets/images/UCSF_Diller_navy_RGB72dpi.png"
        alt="HDFCCC"
        style="margin-left: 16px; margin-bottom: 0;
        width: 230px; height: 0px;">
 </a>
</div>

  <div class="container">
    <h1 class="printonly">UCSF Computational Biology Core (CBC)</h1>
<h1>Software and Computing Resources</h1>

<h2>Our software contributions</h2>

<p>We contribute to several open-source projects including <a href="http://www.r-project.org/">R</a> and
<a href="http://www.bioconductor.org/">Bioconductor</a> and core member Henrik Bengtsson is the creator of the
<a href="http://www.aroma-project.org/">Aroma Project</a>.  Being parts of these projects helps us to keep up to
date with the field and to get invaluable feedback on our own software
and work.</p>

<p>Here is a list of software tools that we have developed ourselves or contributed to their development:</p>

<ul>
<li><a href="http://www.bioconductor.org/packages/release/bioc/html/affxparser.html">affxparser</a>: Software for parsing Affymetrix microarray files.</li>
<li><a href="http://www.aroma-project.org/packages/aroma/">aroma</a>: Preprocessing of spotted microarray data.</li>
<li><a href="http://cran.r-project.org/package=aroma.affymetrix">aroma.affymetrix</a>: Analysis of small to extremely large Affymetrix microarray data sets.</li>
<li>aroma.seq: High-throughput sequence (HT-Seq) analysis in the Aroma framework.  (Pre-release.)</li>
<li><a href="http://cran.r-project.org/package=babel">babel</a>: Ribosome profiling data analysis using statistical
methods with the same name.</li>
<li><a href="http://www.bioconductor.org/packages/release/bioc/html/DNAcopy.html">DNAcopy</a>: Circular Binary Segmentation (CBS) method for aCGH copy
number analysis.</li>
<li><a href="http://akt.ucsf.edu/EGAN/">EGAN</a>: Exploratory Gene Association Networks.</li>
<li>[future]: Asynchroneous (parellel/distributed) processing in R using futures.</li>
<li><a href="http://www.bioconductor.org/packages/release/bioc/html/illuminaio.html">illuminaio</a>: Software for parsing Illumina microarray files.</li>
<li><a href="http://cran.r-project.org/package=matrixStats">matrixStats</a>: Fast and memory-efficient mathematical operations on matrices.</li>
<li><a href="http://cran.r-project.org/package=partDSA">partDSA</a>: Piecewise constant estimation of increasingly complex predictors.</li>
<li><a href="http://cran.r-project.org/package=PSCBS">PSCBS</a>: Parent-specific copy number segmentation using CBS.</li>
<li><a href="http://www.bioconductor.org/packages/release/bioc/html/QDNAseq.html">QDNAseq</a>: Quantitative DNA sequencing for chromosomal aberrations using shallow DNA-Seq.</li>
<li><a href="http://cran.r-project.org/package=R.matlab">R.matlab</a>: R-to-MATLAB connectivity and methods for reading and writing MAT files.</li>
<li><a href="http://cran.r-project.org/package=R.rsp">R.rsp</a>: Dynamic generation of scientific reports for reproducible
research.&nbsp;<sup><a href="#fn1" id="r1">[1]</a></sup></li>
<li><a href="https://r-forge.r-project.org/R/?group_id=349">sfit</a>: Multidimensional simplex fitting.</li>
</ul>

<h2>Software and Reproducible Research</h2>

<p>One of our priorities is to provide scientifically sound and
reproducible research results.  In order to achieve this we make use
of a large number of high-quality computational software tools
provided by either industry or academia.  We try to use
open-source software as much as possible, particularly because it is
key to reproducible research.</p>

<h2>Compute Cluster</h2>

<p>The amount of data being collected in genomic research has grown
dramatically.  It has been less than a decade ago since Affymetrix SNP
array data (~60MB/sample) were considered large.  Many software tools
could handle only 10-20 arrays in multi-sample studies.  This was one
of the reason Henrik Bengtsson developed the <a
href="http://www.aroma-project.org/">Aroma Project</a>, which handles
tens of thousands of arrays even on systems with limited memory
resources.  When high-throughput sequencing
(HT-Seq) entered the arena, there was a paradigm shift in the amount
of data that needed to be processed.  Sequencing the DNA of a <em>single
human genome</em> at 50 times coverage produces a <em>~250GB data file</em> of
aligned reads.  Yes, that is ~4000 times larger file than what we
get with microarray technologies.  (This does <em>not</em> mean that we get
4000 times more &quot;information&quot; from HT-Seq data, but that is a
different story.)  We are now extending the Aroma Project for it to
support HT-Seq analysis as well.</p>

<p>At the UCSF Helen Diller Family Comprehensive Cancer Center we have a
large Linux compute cluster (maintained by <a href="http://ti.ucsf.edu/hpc">Translational
Informatics</a>), which, together with tailored implementations (e.g. our
aroma.seq and aroma.affymetrix tools), allows us to process large
amounts of data in a highly parallel fashion.</p>

<h2>Programming Languages</h2>

<p>We are experienced in programming languages such as C, C++, Java,
Perl, Python and R, to name a few.</p>

<hr>

<div style="font-size: small;">
<em>Footnotes:</em><br>

<span id="fn1"><a href="#r1">[1]</a>
This site was written using RSP-embedded Markdown and compiled with
the R.rsp package.<br></span>
</div>
  </div> 

</div> 

<div id="filler"></div>


<div id="push"></div> 
<div id="footer" style="height: auto;">
  <div class="container">
<p class="muted credit" style="font-size: small;">
<a href="http://cbc.ucsf.edu/">Computational&nbsp;Biology&nbsp;Core</a>
<span class="printonly">(http://cbc.ucsf.edu/)</span>
::
<a href="http://cancer.ucsf.edu/">UCSF&nbsp;Helen&nbsp;Diller&nbsp;Family Comprehensive&nbsp;Cancer&nbsp;Center</a>
::
<a href="http://www.ucsf.edu/">University&nbsp;of&nbsp;California, San&nbsp;Francisco</a>
</p>
  </div>
</div>

<div class="device-xs visible-xs"></div>
<div class="device-sm visible-sm"></div>
<div class="device-md visible-md"></div>
<div class="device-lg visible-lg"></div>

<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.1/jquery.min.js"></script>
<script src="http://netdna.bootstrapcdn.com/bootstrap/3.2.0/js/bootstrap.min.js"></script>
<script>
  $(document).ready(function(){
      $(document.body).css('padding-top', $('#topnavbar').height() + 10);
      $(window).resize(function(){
          $(document.body).css('padding-top', $('#topnavbar').height() + 10);
      });
  });
</script>
<script type="text/javascript">
$(window).load(function() {
  // executes when complete page is fully loaded, including 
  // all frames, objects and images


  // Record location of the beginning of page
  // before displaying the sublogo
  var offsetBefore = 0;
  var first = $("h1:visible:first");
  if (first.length == 0) first = $("h2:visible:first");
  if (first.length > 0) offsetBefore = first.offset().top;

  // Locate the top navigation bar
  var topnavbar = $('#topnavbar');

  // Make sub-logo visible
  var sublogocontainer = $("#sublogocontainer");
  if (sublogocontainer.length > 0) {
    var sublogo = $("#sublogo");
    sublogo.css('height', 'auto');
    sublogo.css('margin-bottom', '5ex');
    sublogocontainer.css('visibility', 'visible');

    // Hide/show main UCSF logo depending on visibility of sub-logo
    if (sublogo.length > 0) {
      var logo = $(".top-banner.logo .inner .logo");
      if (logo.length > 0) {
        logo.css('visibility', 'hidden');
        logo.click(function() {
          window.location.href = "http://www.ucsf.edu/";
        });
        var scrollOffset = 0;
        if (topnavbar.length > 0) scrollOffset = topnavbar.height();
        var sublogoBottom = sublogo.offset().top + 0.5*sublogo.height();

        $(window).scroll(function() {
          if ($(window).scrollTop() + scrollOffset >= sublogoBottom) {
            logo.css('visibility', 'visible');
          } else {
            logo.css('visibility', 'hidden');
          }
        });
      }
    }
  }

  var availableHeight = this.innerHeight;
  if (topnavbar.length > 0) {
    availableHeight = availableHeight - topnavbar.height();
  }
  
  // If sublogo is taking up too much space, scroll page to the beginning
  if (first.length > 0 && sublogocontainer.length > 0 &&
      sublogocontainer.height() > 0.20*availableHeight) {
    var offsetAfter = first.offset().top;
    var shift = offsetAfter - offsetBefore;

    // Do we need to add extra space at the bottom?
    // If not done, the page won't scroll as much as we want
    var pushOffset = $("#push").offset().top;
    var totalHeight = this.innerHeight;
    var available = pushOffset - totalHeight;
    if (available < shift) {
      $("#filler").height(shift - available);
    }

    // Scroll such that the H1 header is at the top
    this.scroll(0, shift);
  }
});
</script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
  
    ga('create', 'UA-56080795-1', 'auto');
    ga('send', 'pageview');
  </script>
</body>
</html>
