# Scientific Software

One of our priorities is to provide scientifically sound and
reproducible research results.   In order to achive this we make use
of a large number of high-quality computational software tools
provided by the industry and the academic world.  We try to use
open-source software as far as possible.


## Our contributions
In addition to being users, we also contribute to several open-source
projects including [R] and [Bioconductor] and core member Henrik
Bengtsson is the creator of the [Aroma Project].  This helps us to keep
up to date with the field, to get invaluable feedback on our own
software and work.

Here is an excerpt of the software tools we have developed:

* [affxparser]: Software for parsing Affymetrix microarray files.
* [aroma]: Preprocessing of spotted microarray data.
* [aroma.affymetrix]: Analysis of small to extremely large Affymetrix microarray data sets.
* [aroma.seq]: High-throughput sequence (HT-Seq) analysis in the Aroma framework.
* [babel]: Ribosome profiling data analysis using the statistical
  method with the same name.
* [DNAcopy]: Circular Binary Segmentation (CBS) method for aCGH copy
  number analysis.
* [EGAN]: Exploratory Gene Association Networks.
* [illuminaio]: Software for parsing Affymetrix microarray files.
* [matrixStats]: Fast and memory-efficient mathmatical operations on matrices.
* [partDSA]: Piecewise constant estimation of increasingly complex predictors.
* [PSCBS]: Parent-specific copy number segmentation using CBS.
* [QDNAseq]: Quantitative DNA sequencing for chromosomal aberrations using shallow DNA-Seq.
* [R.matlab]: R-to-MATLAB connectivity and methods for reading and writing MAT files.
* [R.rsp]: Dynamic generation of scientific reports for reproducible research.
* [sfit]: Multidimensional simplex fitting.


## Programming Languages
We are using several state-of-the-art tools for our daily work.
We are experienced in programming languages such as C, C++,
Java, Perl, Python and R, to name a few.


<%-- PROJECTS --%>
[R]: http://www.r-project.org/
[Bioconductor]: http://www.bioconductor.org/
[Aroma Project]: http://www.aroma-project.org/

<%-- SOFTWARE --%>
[affxparser]: http://www.bioconductor.org/packages/release/bioc/html/affxparser.html
[aroma]: http://www.aroma-project.org/packages/aroma/
[aroma.affymetrix]: http://cran.r-project.org/package=aroma.affymetrix
[aroma.seq]: http://aroma-project.org/alpha/aroma.seq/
[babel]: http://cran.r-project.org/package=babel
[DNAcopy]: http://www.bioconductor.org/packages/release/bioc/html/DNAcopy.html
[EGAN]: http://akt.ucsf.edu/EGAN/
[illuminaio]: http://www.bioconductor.org/packages/release/bioc/html/illuminaio.html
[matrixStats]: http://cran.r-project.org/package=matrixStats
[partDSA]: http://cran.r-project.org/package=partDSA
[PSCBS]: http://cran.r-project.org/package=PSCBS
[QDNAseq]: http://www.bioconductor.org/packages/release/bioc/html/QDNAseq.html
[R.matlab]: http://cran.r-project.org/package=R.matlab
[R.rsp]: http://cran.r-project.org/package=R.rsp
[sfit]: https://r-forge.r-project.org/R/?group_id=349
